src/util/init_logging.cpp

```diff
@@ -43,239 +42,223 @@
+            {
+                util::format_to(to, "{:08x}", locality_id);
             }
-            else {
+            else
+            {
                 // called from outside a HPX thread
-                str.prepend_string(std::string(sizeof(std::uint32_t)*2, '-'));
+                to << std::string(8, '-');
             }
         }
     };
 
     ///////////////////////////////////////////////////////////////////////////
     // custom formatter: HPX thread id
-    struct thread_id
-      : hpx::util::logging::formatter::class_<thread_id>
+    struct thread_id : logging::formatter::manipulator
     {
-        void operator()(param str) const
+        void operator()(std::ostream& to) const override
         {
             threads::thread_self* self = threads::get_self_ptr();
-            if (nullptr != self) {
+            if (nullptr != self)
+            {
                 // called from inside a HPX thread
                 threads::thread_id_type id = threads::get_self_id();
-                if (id != threads::invalid_thread_id) {
-                    std::stringstream out;
-                    out << std::hex << std::setw(sizeof(void*)*2)
-                        << std::setfill('0')
-                        << reinterpret_cast<std::ptrdiff_t>(id.get());
-                    str.prepend_string(out.str());
+                if (id != threads::invalid_thread_id)
+                {
+                    std::ptrdiff_t value =
+                        reinterpret_cast<std::ptrdiff_t>(id.get());
+                    util::format_to(to, "{:016x}", value);
                     return;
                 }
             }
 
             // called from outside a HPX thread or invalid thread id
-            str.prepend_string(std::string(sizeof(void*)*2, '-'));
+            to << std::string(16, '-');
         }
     };
 
     ///////////////////////////////////////////////////////////////////////////
     // custom formatter: HPX thread phase
-    struct thread_phase
-      : hpx::util::logging::formatter::class_<thread_phase>
+    struct thread_phase : logging::formatter::manipulator
     {
-        void operator()(param str) const
+        void operator()(std::ostream& to) const override
         {
             threads::thread_self* self = threads::get_self_ptr();
-            if (nullptr != self) {
+            if (nullptr != self)
+            {
                 // called from inside a HPX thread
                 std::size_t phase = self->get_thread_phase();
-                if (0 != phase) {
-                    std::stringstream out;
-                    out << std::hex << std::setw(sizeof(std::uint32_t))
-                        << std::setfill('0') << self->get_thread_phase();
-                    str.prepend_string(out.str());
+                if (0 != phase)
+                {
+                    std::string out =
+                        format("{:04x}", self->get_thread_phase());
```

Comment:
This should probably be output somewhere?

Reply:
I have a vague memory of you already fixing this. Did this maybe get reverted at some point?

Reply:
I remember that too.. maybe it was a different instance? or perhaps something get lost during a rebase